### DISCUSSION TITLE
What's the purpose of having both an array and a mapping?

### DISCUSSION THREAD

**Discussion by cryptografiq** [Cyfrin/foundry-full-course-cu#1519] [Q&A]
I'm on section 3.12 of the solidity course and I'm curious why he set up both an array and a mapping for funders. Isn't the mapping sufficient? Is it standard practice to always have an array when you have a mapping? Thanks!

<img width="1026" alt="Screen Shot 2024-03-23 at 11 09 27 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/103517626/dd28b131-12ba-42c7-bbcf-96337875ad46">

**Comment by EngrPips** [comment]
Hello @cryptografiq , what an interesting question you asked. Ordinarily it is enough for us to just have an array where we can push every address that funds our contract but because we want to keep track of the amount of ether that each address sends to us we needed a mapping as an array isn't enough for such requirement. Hence that is why we have an array and a mapping. The bottom line is we have a mapping because we want to know how much each address send to us.

I hope my explanation is clear enough to help you understand
